app/forms/declaration/GoodsLocation.scala

```diff
@@ -19,55 +19,53 @@ package forms.declaration
 import forms.DeclarationPage
 import play.api.data.Forms.{optional, text}
 import play.api.data.{Form, Forms}
-import play.api.libs.json.Json
+import play.api.libs.json.{Json, OFormat}
 import services.Countries.allCountries
 import utils.validators.forms.FieldValidator._
 
-case class GoodsLocation(
-  typeOfLocation: String,
-  qualifierOfIdentification: String,
-  identificationOfLocation: Option[String],
-  additionalIdentifier: Option[String],
-  addressLine: Option[String],
-  city: Option[String],
-  postCode: Option[String],
-  country: String
-)
+case class GoodsLocation(code: String)
```

Comment:
Would making the 'code' optional make the data migration easier?  In fact, you might not have to migrate the data at all.  Unless you are planning on synthesising the code based on the contents of the existing fields.

Reply:
Unfortunately not, because previously we had a lot of fields, I'll try to achieve it in easy way, but if not happy to discuss other approaches

Reply:
Based on the xml fields I stay with the backend model, so just changed the form model